src/main/java/org/galatea/starter/entrypoint/IexRestController.java

```diff
@@ -76,29 +81,44 @@ public List<IexLastTradedPrice> getLastTradedPrice(
   @GetMapping(value = "${mvc.iex.getHistoricalPricePath}", produces = {
       MediaType.APPLICATION_JSON_VALUE})
   public IexHistoricalPrice getHistoricalPrice(
-      @RequestParam(value = "symbol") @NotNull final String symbols,
+      @RequestParam(value = "symbol") @NotNull @NotEmpty final String symbols,
       @RequestParam(value = "range", required = false) final Optional<@Pattern.List(
           {@Pattern(regexp = ".*"), @Pattern(regexp = VALID_DATES)}) String> range,
       @RequestParam(value = "date", required = false) final Optional<Integer> date) {
```

Comment:
While the format of this param happens to be all numbers, it's not _really_ an integer - it's a numeric string (as in, a string of characters that all happen to be numbers). Like when someone passes in 20221019 it doesn't mean "20 million, 210 thousand, 19", it's just a representation of a date that has no separators. Treating it as an integer is slightly misleading as to what the variable holds, potentially makes working with the variable a bit harder (e.g. in `LocalDate provided = LocalDate.parse("" + dateVal, formatter);` below you need to convert it to a String first, and if we wanted validation on this date value we'd need to that there as well), and would require more effort if - for whatever reason - the date format changed in the future to something with separators.
It's also something to be careful of for other types of numeric strings, especially ones that can have leading 0s (e.g. if you put the account number "000123" into an int you could end up with some issues down the line).
It would be better to take it in as a string here, or even as a LocalDate, since both of those more accurately represent what the data _is_ and make it easier to perform logic on the field.

Reply:
I mainly had it as an integer because I needed 3 separate methods in IexClient for the different mappings, as I couldn't have two methods with the same name and parameters (String, String). Since you are right that date is better handled as a String, I am simply renaming one of the (String, String) methods to fix this.

I could take it in as a LocalDate, but would then need to eventually convert it into a String to be passed into the IexClient method anyway (already formatted to be used in the Iex API call), and it is easier to store into different types of databases as a String